### PR TITLE
Accept "preserve" family of rsync arguments [RHELDST-6172]

### PR DESCRIPTION
**Pull request by nathanegillett**
This commit updates the list of ignored arguments with rsync's "preserve"
family of arguments, such as "--hard-links" to preserve hard links.
These arguments, like other ignored arguments, are passed through to
standard rsync when not using exodus-gw.

### CODE REVIEW

File: internal/args/parse.go [hunk 1]
```diff
@@ -24,18 +24,30 @@ func (f filterArgument) Validate() error {
...
+	HardLinks       bool `short:"H"`
+	Perms           bool `short:"p"`
+	Executability   bool `short:"E"`
+	Acls            bool `short:"A"`
+	Xattrs          bool `short:"X"`
+	Owner           bool `short:"o"`
+	Group           bool `short:"g"`
+	Devices         bool
+	Specials        bool
+	DevicesSpecials bool   `short:"D" hidden:"1"`
```
Comment (nathanegillett) [inline internal/args/parse.go:40]: As mentioned in the commit message body, I wasn't able to find a practical way to flag both Devices and Specials using "-D" so it's wholly ignored. Apparently multiple arguments cannot share a flag, the first one parsed wins and the other ignored.

Comment (rohanpm) [reply internal/args/parse.go:40]: I don't think this is quite right - I mean I get that maybe kong library doesn't have a trivial way to do this, but I don't think that's a good reason to just give up on passing this through to rsync.

Can't we do this with a couple of lines of code just after we call kong.Parse: if out.DevicesSpecials is true, then we set out.Devices true and out.Specials true.  There shouldn't be any problem then passing --devices and --specials through to rsync.  We would not literally be passing through the `-D` argument, instead we'd be "expanding" it, but I think that's OK.

**Review by rohanpm — CHANGES_REQUESTED** [review]
I think this change is only half done - this is the part where we accept arguments, but what about the part where we feed them into rsync?

See internal/rsync/rsync.go.  It reassembles an rsync command-line based on an args.Config object, I think you would need to add branches for all the new arguments here too (unless you can come up with a smarter way to do it).

File: internal/args/parse.go (hunk 1 above)
Comment (nathanegillett) [reply internal/args/parse.go:40]: OK; done. I wasn't sure if this one flag was worth the additional, albeit minute, code.

**Review by rohanpm — APPROVED** [review]

**Review by crungehottman — APPROVED** [review]

**Comment by rohanpm** [comment]
@nathanegillett , do you want to try doing a release for this via https://github.com/release-engineering/exodus-rsync/wiki/Releasing ?  It hasn't been used yet since my own initial project setup, would be nice to verify it's working under normal development conditions.

**Pull request merged**